apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -0,0 +1,313 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Spanner.V1;
+
+// ReSharper disable UnusedParameter.Local
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// A SpannerConnection represents a connection to a single Spanner database.
+    /// </summary>
+    public class SpannerConnection : DbConnection
+    {
+        private SpannerBatchOperation _activeBatchOperation;
```

Comment:
you *can* actually change the connection string or database -- which is kind of a PITA, but in my currently implementation I require that the connection be closed to simplify that.
so short answer is -- not much can be made readonly here, but as I progress resharper will be screaming at me to make those that can be and I will follow suit.